### DISCUSSION TITLE
Dry-out During CO2 Injection

### DISCUSSION THREAD

**Discussion by denismart997** [idaholab/moose#21660] [Q&A Modules: General] [answered]
Dear, Moose community

My name is Denis , I was exploring different Moose applications for CO2 storage, went through couple papers, such as (https://doi.org/10.3929/ethz-b-000410774),  "A Numerical Model for Formation Dry-out During CO2 Injection…”, and currently I am a little bit confused , for modelling such behaviour - geochemical system should be considered (porous flow tutorial 13) , including reactions + hard coded salt precipitation when the solubility limit is exceed. Or everything is much simpler , and it is possible to run simple CO2 injection case with nonlinear NaCl variable. However, after simulations in this case only salt transport is reflected, without dry-out and salt precipitation around injection point.  (By mentioning dry-out front I mean the continuous injection of dry scCO2 vaporizes the remaining formation brine in the swept zone, which persistently increases the concentration of dissolved salt. When the salt concentration reaches the solubility limit, it precipitates out of the brine). Thank you in advance.

Kind regards,
Denis.

**Comment by WilkAndy** [comment, marked as answer]
@cpgr is the world expert in this type of problem

**Reply by cpgr** [reply]
@WilkAndy is overstating my expertise, but I think I can explain this paper.

Like you say, as you inject CO2, the brine near the well gets pushed out until it is immobile (liquid saturation below its residual saturation). This immobile water continues to evaporate into the gas (or supercritical) phase, and is transported away. The NaCl in the brine will then start to precipitate once the amount of NaCl is above it's equilibrium concentration.

If you want to model this properly, you can do a coupled geochemistry - flow simulation like you note. In the linked study, the authors don't explicitly include a solid NaCl phase once the mass fraction of salt exceeds its equilibrium concentration due to evaporation of the water. Instead, they reduce the fracture aperture (they are modelling flow into a fracture only) using a simple model to account for the reduction due to precipitate. They then reduce the amount of NaCl in the liquid phase by resetting the mass fraction of NaCl to its equilibrium value in each cell.

There are more details in the corresponding journal article https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/2020WR027213

**Reply by denismart997** [reply]
Dear, @cpgr

Thank you very much for your reply and clarification. Could you please provide one more hint , about brineco2 module in moose? Mainly , I have a question regarding a local equilibrium phase partitioning and mass fraction of water in the CO2 phase. As I understood you are applying the same approach as it is done in Tough2 simulator (ECO2N module), where local equilibrium is calculated according to its solubility limit at a certain pressure, temperature and salinity (Spycher and Pruess, 2005), and this
implies that phase partitioning occurs instantaneously? Am I right? If yes , from your point of you, can it be done differently, using kinetic approach? The example is this paper "(http://dx.doi.org/10.1016/j.ijggc.2014.05.010)".  Thank you in advance.

Kind regards,
Denis.

**Reply by cpgr** [reply]
That's correct - we use an instantaneous equilibrium assumption (same as TOUGH). We don't have the capability to compute the evaporation of water into the gas phase using a kinetic approach currently. However, I have previously hacked together something for the related problem of kinetic dissolution of the gas phase into the water phase. I can't remember the fine details off the top of my head, but an example of the difference between instantaneous dissolution of CO2 into the water and dissolution computed using a kinetic mass transfer with MOOSE is (for a 1D radial model of flow from a well)

![xco2l_eqvkd](https://user-images.githubusercontent.com/7664823/182049256-c60e8402-ca44-4b07-897f-43d65e92469f.png)

So it is certainly possible to do what you want with a bit of coding.

**Reply by denismart997** [reply]
Dear, @cpgr

Thank you very much for you reply regarding last question . Sorry , for disturbing you once again, for my master thesis instantaneous equilibrium approach should be sufficient, so I want to repeat something similar to the simulation which was performed from the first (aperture) paper , but much simpler - just scCO2 core flooding experiment, slightly heterogeneous with random porosity distribution just for visualisation. However, since there is no input file available from that experiment (only output .e file), I have problems with injection boundary, I have tried simple PorousFlowSink , but it requires significant injection rates (1000 more than it should be) just to let fluid enter the core(even if Pc=0), and after 3 sec. of simulation there are some convergence issues. I have tried DircacKernel point injection, and it works much better , but I need injection to be performed from the whole inflow boundary. If you have couple min, could you please have a look into input file , might something catch you attention . Thank you very much!
https://github.com/denismart997/MooseCO2/blob/b0ed0e37073de5e6058c6ad352d721dcdeb903cf/CO2_StartS2.i
<img width="618" alt="Point_Inj" src="https://user-images.githubusercontent.com/105974820/184248247-e028e3a5-5447-41c2-a6a5-2c3b715a87ed.png">
<img width="609" alt="Point_Inj1" src="https://user-images.githubusercontent.com/105974820/184248269-667ed162-0105-45d5-a033-78d6a9be06d8.png">

**Reply by cpgr** [reply]
The input file you linked didn't really work as the initial residual was lower than your `nl_abs_tol`, so nothing happens.

After some minor tweaks I could get it to run and it seems to do something sensible (but I don't know if the injection rate is what you want)
![denisco2](https://user-images.githubusercontent.com/7664823/184261218-d0866499-e793-41de-a1a7-1eaf674f6568.png)

What I did was:

- remove the scaling of the pressure variable
- set `nl_abs_tol = 1e-11`

and I could get the above.

If you add the following to your input file
```
[Debug]
  show_var_residual_norms = true
[]
```
you can see that the residuals of the unscaled nonlinear variables are all around the same magnitude.

You might also want to add a post processor to check that you are injecting the exact amount of CO2 that you expect
```
[Postprocessors]
  [co2mass]
    type = PorousFlowFluidMass
    phase = '0 1'
    fluid_component = 1
    execute_on = 'initial timestep_end'
  []
[]
```
will compute the CO2 mass (in kg) in all phases.

Finally, you could remove the `bottomnaclg` BC as there is no NaCl in the gas phase, so this is always 0.

Let me know how it goes,
Chris

**Reply by denismart997** [reply]
Dear , @cpgr

Thank you very much, Chris, for you help , it look much better know . This is what I ve got after 3000 sec , I thought that salt concentration would be more prominent by this stage, as it was at single point injection. But anyway looks much better. Thanks a lot!
<img width="563" alt="Screen Shot 2022-08-13 at 1 09 10 am" src="https://user-images.githubusercontent.com/105974820/184455795-d5fa39d0-c0ae-4ea8-b41a-dc9e8090ef5c.png">
<img width="517" alt="Screen Shot 2022-08-13 at 1 09 24 am" src="https://user-images.githubusercontent.com/105974820/184455821-3734684c-0a52-4331-91da-4538d5c1af6d.png">

**Comment by robo-boluo** [comment]
Dear Chris Green and and guillaumeGiudicelli  @cpgr @GiudGiud ,

May I ask if there are examples of CO2 injection into brine containing multiple salts, such as NaCl and CaCl2, MgCl2...? Additionally, if the kernels (PorousFlowMassTimeDerivative, PorousFlowAdvectiveFlux) are utilized to simulate CO2, water, and brine flow, how can the mass fractions of ions (f_H, f_Na, f_Ca, f_HCO3....) be saved? These saved fraction variables are necessary for coupling into the geochemistry model.
Alternatively, can the [PorousFlowFullySaturated] action be employed to address CO2 injection into brine, facilitating the direct preservation of mass fractions using save_component_rate_in? I gleaned information on the usage of PorousFlowFullySaturated from the reactive transport example located in ['moose/modules/combined/examples/geochem-porous_flow/forge/porous_flow.i']."

Appreciate your time and guidance on this matter.

Regards,

Robo

**Reply by cpgr** [reply]
I don't believe there are any examples of this. The kernels etc can handle multiple fluid components. How difficult this is to do depends on whether you want the CO2 in a gas phase to be able to dissolve into the fluid.

**Reply by robo-boluo** [reply]
Hi Chris Green @cpgr 
Thank you for your kind reply. For practical application, the injected CO2 is anticipated to exist in both gas and liquid phases. The dissolved CO2 will undergo a reaction with H2O, yielding HCO3- and H+ ions. Subsequently, the H+ ions will be utilized for reactive transport modeling, including mineral dissolution and precipitation. I'm not certain about the feasibility of employing MOOSE to model this scenario
